tests/multiprocess_gpu_test.py

```diff
@@ -38,8 +45,37 @@
 except ImportError:
   pytest = None
 
+config.update("experimental_xmap_spmd_lowering", True)
 config.parse_flags_with_absl()
 
+def sorted_devices():
+  devices = sorted(jax.devices(), key=lambda d: (d.id, d.host_id))
+  if len(devices) != 16:
+    raise unittest.SkipTest("Test assumes that it runs on 16 devices (2 nodes)")
+  return devices
+
+def create_2d_non_contiguous_mesh():
```

Comment:
Make these methods of Slurm test class. No need for this to exist outside of that class.